src/sync/sync_user.hpp

```diff
@@ -53,16 +57,150 @@ struct SyncUserIdentifier {
     }
 };
 
+// A struct that decodes a given JWT.
+struct RealmJWT {
+    // The token being decoded from.
+    std::string token;
+
+    // When the token expires.
+    long expires_at;
+    // When the token was issued.
+    long issued_at;
+    // Custom user data embedded in the encoded token.
+    util::Optional<std::map<std::string, util::Any>> user_data;
+
+    RealmJWT(const std::string& token);
+
+    bool operator==(const RealmJWT& other) const
+    {
+        return token == other.token;
+    }
+};
+
+struct SyncUserProfile {
```

Comment:
There doesn't really appear to be any reason for this to not just be a simple struct that's passed around by value rather than a shared object which then needs getters and private members.